File: python_scripts/script_1.py
#!/usr/bin/env python
'''
	python_scripts/script_1.py
	--------------------------------------------------------
	This script is a working file for this tutorial. You
	will need to edit this file to implement reviewer
	comments.

	You may run this script individually via

		$ python python_scripts/script_1.py
'''


def drawU():
	'''Draw the U body'''
	# read in data
	U = pd.read_csv('data/letter_1.csv',
					header = None)
	x, y = list(U[0]), list(U[1])
	x.append(x[0])
	y.append(y[0])

	# plot data
	ax.fill(x, y, color = 'black')
	ax.plot(x, y, color = 'black', linewidth = 2)

# Solution
# def drawU():
# 	'''Draw the U body'''
# 	# read in data
# 	U = pd.read_csv('data/letter_1.csv',
# 					header = None)
# 	x, y = list(U[0]), list(U[1])
# 	x.append(x[0])
# 	y.append(y[0])
#
# 	# plot data
# 	ax.fill(x, y, color = (1, 0.8, 0))  # per reviewer comment
# 	ax.plot(x, y, color = 'black', linewidth = 2)


# You will not need to edit the code below
if __name__ == '__main__':
	# We are running this script on its own
	import pandas as pd
	import matplotlib.pyplot as plt
	fig, ax = plt.subplots(1, 1)		
	ax.axis('off')
	drawU()
	plt.show()
else:
	# We are running this script from within main_script.py
	# Thus all the dependencies and globals have already 
	# been imported.
	drawU()

File: R_scripts/script_1.R
#!/usr/bin/env Rscript
#
# R_scripts/script_1.R
# --------------------------------------------------------
# This script is a working file for this tutorial. You
# will need to edit this file to implement reviewer
# comments.
#
# You can run this script "individually" by 
# calling the `test_U` function defined below.
#

library(ggplot2)

draw_U <- function(canvas) {
  # Adds a letter "U" to an existing ggplot2 context
  
  # read in data
  U <- read.csv(file = 'data/letter_1.csv', 
                    header = FALSE)
  names(U) <- c('x', 'y')
  U[nrow(U) + 1,] <- U[1,]
  
  # plot data
  canvas + 
    geom_polygon(
      data=U,
      fill='black', 
      color='black', 
      size=1, 
      aes(x=x, y=y))     
}


test_U <- function() {
  # Displays just the U body
  
  canvas <- ggplot(NULL) + 
    theme(axis.title=element_blank(),
          axis.text=element_blank(),
          axis.ticks=element_blank(),
          panel.background=element_blank(),
          panel.grid=element_blank())
  
  draw_U(canvas)  
}

# Solution
# draw_U <- function(canvas) {
#   # Adds a U body to an existing ggplot2 context
# 
#   # read in data
#   U <- read.csv(file = 'data/U.csv',
#                     header = FALSE)
#   names(U) <- c('x', 'y')
#   U[nrow(U) + 1,] <- U[1,]
# 
#   # plot data
#   canvas +
#     geom_polygon(
#       data=U,
#       fill=rgb(1, 0.8, 0),  # per reviewer comment
#       color='black',
#       size=1,
#       aes(x=x, y=y))
# }
